warehouse/slave.go

```diff
@@ -101,40 +119,58 @@ func (job *PayloadT) getFileManager() (filemanager.FileManager, error) {
 
 /*
  * Download Staging file for the job
+ * If error occurs with the current config and current revision is different from staging revision
+ * We retry with the staging revision config if it is present
  */
 func (jobRun *JobRunT) downloadStagingFile() error {
-	filePath := jobRun.stagingFilePath
-	file, err := os.Create(filePath)
-	if err != nil {
-		panic(err)
-	}
-
 	job := jobRun.job
-	downloader, err := jobRun.job.getFileManager()
-	if err != nil {
-		pkgLogger.Errorf("[WH]: Failed to initialize downloader")
-		return err
-	}
+	downloadTask := func(config interface{}, useRudderStorage bool) (err error) {
```

Comment:
Good job with using the inline function. Very concise and clear 🎉 🎈

Reply:
TYSM.